com.civicrm.hrjobroles/js/src/job-roles/services/date-validation.js

```diff
@@ -39,6 +39,36 @@ define([
             }
         }
 
+        function checkIfContractStartDateIsLowerThanStart (contract_start, start_date) {
+            if (contract_start.isAfter(start_date)) {
+                _error('Start Date cannot be lower than Contract Start Date.', ['start_date']);
+            }
+        }
+
+        function checkIfStartIsLowerThanContractEnd (start_date, contract_end) {
+            if (start_date.isAfter(contract_end)) {
+                _error('Start Date cannot be higher than Contract End Date.', ['start_date']);
+            }
+        }
+
+        function checkIfEndIsEqualOrLowerThanContractEnd (end_date, contract_end) {
+            if (end_date.isAfter(contract_end)) {
+                _error('End Date cannot be higher than Contract End Date.', ['end_date']);
+            }
+        }
+
+        function formatDate(date) {
```

Comment:
What's the reasoning here (again, document the functions)? If `date` is a `Date` object then sets its time to midnight, and return its timestamp, but otherwise don't bother? Does it mean that if `date` is not a `Date`, then we take for granted it's a timestamp set to midnight?